From cd93aaa2e096a8cbd1f1789dcce06857067b35c9 Mon Sep 17 00:00:00 2001
From: Brian Harring <ferringb@gmail.com>
Date: Mon, 11 Dec 2023 17:10:20 -0800
Subject: [PATCH] fix(systemd-255): handle systemd-pcr{phase -> extend} rename

The binary systemd-pcrphase was renamed to systemd-pcrextend
in systemd 255, but the backing units were all left named
systemd-pcrphase.

Fixes: #2583

Signed-off-by: Brian Harring <ferringb@gmail.com>
---
 modules.d/01systemd-pcrphase/module-setup.sh | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/modules.d/01systemd-pcrphase/module-setup.sh b/modules.d/01systemd-pcrphase/module-setup.sh
index fa960a42c1..87efd0c1a6 100755
--- a/modules.d/01systemd-pcrphase/module-setup.sh
+++ b/modules.d/01systemd-pcrphase/module-setup.sh
@@ -6,7 +6,11 @@
 check() {

     # If the binary(s) requirements are not fulfilled the module can't be installed.
-    require_binaries "$systemdutildir"/systemd-pcrphase || return 1
+    # systemd-255 renamed the binary, check for old and new location.
+    if ! require_binaries "$systemdutildir"/systemd-pcrphase && \
+       ! require_binaries "$systemdutildir"/systemd-pcrextend; then
+       return 1
+    fi

     # Return 255 to only include the module, if another module requires it.
     return 255
@@ -28,6 +32,7 @@ install() {

     inst_multiple -o \
         "$systemdutildir"/systemd-pcrphase \
+        "$systemdutildir"/systemd-pcrextend \
         "$systemdsystemunitdir"/systemd-pcrphase.service \
         "$systemdsystemunitdir/systemd-pcrphase.service.d/*.conf" \
         "$systemdsystemunitdir"/systemd-pcrphase-sysinit.service
